compiler/codegen/codegen.cpp

```diff
@@ -1117,12 +1117,17 @@ static void codegen_header_compilation_config() {
   if (cfgfile.fptr != NULL) {
     FILE* save_cfile = gGenInfo->cfile;
 
-    gGenInfo->cfile = cfgfile.fptr;
+    fprintf(cfgfile.fptr, "/*** %s ***/\n", "Compilation Info");
+    fprintf(cfgfile.fptr, "\n#include <stdio.h>");
+    fprintf(cfgfile.fptr, "\n#include \"chpltypes.h\"\n\n");
 
-    genComment("Compilation Info");
-
-    fprintf(cfgfile.fptr, "\n#include <stdio.h>\n");
-    fprintf(cfgfile.fptr, "\n#include \"chpltypes.h\"\n");
+    if (llvmCodegen && 0 == strcmp(CHPL_LAUNCHER, "none")) {
```

Comment:
Let's put a comment before this conditional explaining that we need the .c file if the launcher will build with it.
Also I see a bunch of other cases with the same conditional. Could you please do something like
```
bool genConfigInLLVM = llvmCodegen && 0 == strcmp(CHPL_LAUNCHER, "none");
if (genConfigInLLVM) ...
```

so that when this logic changes it is not hard to update?